openstates/az/bills.py

```diff
@@ -163,33 +164,35 @@ def scrape(self, chamber, session):
         page = self.get(bill_list_url, cookies=req.cookies).content
         # There's an errant close-comment that browsers handle
         # but LXML gets really confused.
-        page = page.replace('--!>', '-->')
+        page = page.replace(b'--!>', b'-->')
         page = html.fromstring(page)
 
         bill_rows = []
-        if(chamber == 'lower'):
-            bill_rows = page.xpath('//div[@name="HBTable"]//tbody//tr')
-            for row in bill_rows:
-                bill_id = row.xpath('td/a/text()')[0]
-                self.scrape_bill(chamber, session, bill_id)
-
-        elif (chamber == 'upper'):
-            bill_rows = page.xpath('//div[@name="SBTable"]//tbody//tr')
-            for row in bill_rows:
-                bill_id = row.xpath('td/a/text()')[0]
-                self.scrape_bill(chamber, session, bill_id)
-
-        #TODO: MBTable - Non-bill Misc Motions?
+        chambers = [chamber] if chamber else ['upper', 'lower']
+        for chamber in chambers:
+            if(chamber == 'lower'):
+                bill_rows = page.xpath('//div[@name="HBTable"]//tbody//tr')
+                for row in bill_rows:
+                    bill_id = row.xpath('td/a/text()')[0]
+                    yield from self.scrape_bill(chamber, session, bill_id, session_id)
+
+            elif (chamber == 'upper'):
```

Comment:
These two branches look almost identical. Could we refactor to avoid duplicated code?